### DISCUSSION TITLE
Combine multiple images into pdf on MacOS

### DISCUSSION THREAD

**Discussion by shallow-beach** [ImageMagick/ImageMagick#7060] [Help] [answered]
### ImageMagick version

7.1.1-26

### Operating system, version and so on

MacOS 14.2.1 w/ ARM

### Description

This [question](https://stackoverflow.com/q/77857353/22673230) was not received well on StackOverflow, so I am guessing the answer must be obvious.

On Windows, I could combine files named `1.jpg`, `2.jpg`, etc into a file `scan.pdf` using

`magick %d.jpg[1-n] scan.pdf`

where `n` is the total number of jpgs. On MacOS, I tried this and received an error. If I instead try

`magick *.jpg scan.pdf`

it does not render in the correct order. Additionally, attempting

`magick [1-n].jpg scan.pdf`

doesn't seem to work for `n` > 9. What is the correct way to combine multiple consecutively named files in ImageMagick on MacOS (and presumably Linux)?

Simple solutions include the comment from the SO thread linked above, as well as [this answer](https://github.com/ImageMagick/ImageMagick/discussions/6116#discussioncomment-5158321) from another user on github. That is:

`magick $(ls *.jpg | sort -n) scan.pdf`

or

`magick @filelist.txt scan.pdf`

I am wondering if there is a specific method recommended by ImageMagick for this use case of ordered input files, given the above two solutions involve a little shell magic.

**Comment by fmw42** [comment]
Try

`magick *.jpg[1-n] scan.pdf`

Does that work?

**Reply by shallow-beach** [reply]
this returns 'no matches found' for me

**Comment by fmw42** [comment]
What are your filenames?  Are they all the same?  Sorry, I suspect you cannot use wild card input and scene range [1-n] on  Unix.  

You can give a given file name and add [1-n] such as x.TIFF[1-n] and convert to x.pdf.  Or you can give *.jpg and convert to x.pdf.  On unix, the wild card gives images in alphabetic order.

**Reply by shallow-beach** [reply]
i think i may be using that command incorrectly then if it is about the number of pages _within_ a jpg file? my files are laid out like

```
1.jpg
2.jpg
3.jpg
.
.
.
265.jpg
```

and each is just a single page, one photo. 

the comment on alphabetical order explains why it jumps straight to 100.jpg after 1.jpg when i use the *.jpg method!

**Comment by fmw42** [comment]
try

`magick *.jpg miff:- | magick -[1-n] scan.pdf`

Does that work?

**Reply by shallow-beach** [reply]
`no matches found: -[1-n]` where `n` is substituted for any number, on my zsh terminal

**Comment by shallow-beach** [comment, marked as answer]
needed quotes (' ') around the wildcards that i originally used on Windows to work on MacOS, i.e.:

`magick '%d.jpg[1-n]' scan.pdf`

thanks [steeldriver on SO](https://unix.stackexchange.com/q/767525/597701).
